client/client_test.go

```diff
@@ -549,7 +595,25 @@ func TestConcurrentIncrements(t *testing.T) {
 		if err := db.DelRange(testUser+"/value-0", testUser+"/value-1x"); err != nil {
 			t.Fatalf("%d: unable to clean up: %v", k, err)
 		}
-		concurrentIncrements(db, t)
+		concurrentIncrements(db, t, false /*isMultiphase*/)
+	}
+}
+
+// TestMultiphaseTransactions is a simple explicit test for serializability
+// while running multiphase trasactions.
+func TestMultiphaseTransaction(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	s := server.StartTestServer(t)
+	defer s.Stop()
+	db := createTestClient(s.ServingAddr())
+
+	// Convenience loop: Crank up this number for testing this
+	// more often. It'll increase test duration though.
+	for k := 0; k < 5; k++ {
+		if err := db.DelRange(testUser+"/value-0", testUser+"/value-1x"); err != nil {
+			t.Fatalf("%d: unable to clean up: %v", k, err)
+		}
+		concurrentIncrements(db, t, true /*isMultiphase*/)
```

Comment:
I think so. With the transaction state being copied around, and multiple calls using new Txn objects, I feel that this mechanism is very slippery and is best tested in this kind of test where any of the calls can fail and yet we maintain serializability. Every time someone creates a new driver they'll complain about it being unable to achieve serializability and I must prefer us saying we are testing it and here's the test. I also much prefer it being here than in driver_test (language specific).

A case can be made for not doing the non-multiphase part of the test. Perhaps we can eliminate that code path.